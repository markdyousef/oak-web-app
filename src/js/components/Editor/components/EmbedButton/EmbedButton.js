import React, { Component, PropTypes } from 'react';
import { AtomicBlockUtils } from 'draft-js';
import Icon from '../../icons/embed';
import ActionIcon from '../ActionIcon';

export default class EmbedButton extends Component {
    static propTypes = {
        title: PropTypes.string.isRequired,
        editorState: PropTypes.shape({
            _immutable: PropTypes.object
        }).isRequired,
        setEditorState: PropTypes.func.isRequired,
        close: PropTypes.func.isRequired
    }
    onClick = () => {
        const { close } = this.props;
        const url = window.prompt('Enter URL', 'https://www.clai.io');
        close();

        if (!url) return;
        this.addEmbedURL(url);
    }
    addEmbedURL = (url) => {
        const { setEditorState, editorState } = this.props;
        const contentState = editorState.getCurrentContent();
        const entityKey = contentState
            .createEntity('embed', 'IMMUTABLE', { url })
            .getLastCreatedEntityKey();

        setEditorState(
            AtomicBlockUtils.insertAtomicBlock(
                editorState,
                entityKey,
                'E'
            )
        );
    }
    render() {
        const { title } = this.props;
        return (
            <ActionIcon onClick={this.onClick}>
                <Icon />
                {title}
            </ActionIcon>
        );
    }
}
