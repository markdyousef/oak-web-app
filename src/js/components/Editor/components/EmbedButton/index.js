import EmbedButton from './EmbedButton';

export default EmbedButton;
