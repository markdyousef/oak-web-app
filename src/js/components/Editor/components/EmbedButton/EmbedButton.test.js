import React from 'react';
import renderer from 'react-test-renderer';
import EmbedButton from './EmbedButton';

it('renders component correctly', () => {
    const props = {
        title: '',
        editorState: {},
        setEditorState: () => {},
        close: () => {}
    };
    const component = renderer.create(<EmbedButton {...props} />);
    expect(component).toMatchSnapshot(component);
});
