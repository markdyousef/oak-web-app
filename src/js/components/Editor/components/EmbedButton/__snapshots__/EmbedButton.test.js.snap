exports[`[object Object] 1`] = `
<button
  className="hsTvfh"
  onClick={[Function]}>
  <svg
    height="16"
    viewBox="0 0 16 16"
    width="16">
    <g>
      <path
        d="M14.45,6.4l-3.31,3.31c-0.69,0.69-1.6,1.04-2.5,1.04c-0.14,0-0.28-0.01-0.42-0.03c-0.73-0.08-1.44-0.4-2-0.96 C6.06,9.59,5.91,9.41,5.78,9.21C5.63,8.98,5.69,8.67,5.93,8.52c0.23-0.15,0.54-0.08,0.69,0.15c0.09,0.14,0.19,0.26,0.31,0.38 c0.63,0.63,1.53,0.83,2.35,0.61C9.7,9.56,10.11,9.33,10.44,9l3.3-3.3c0.11-0.11,0.2-0.22,0.29-0.34c0.68-0.98,0.61-2.32-0.24-3.17 c-0.46-0.46-1.06-0.7-1.74-0.7c-0.67,0.01-1.3,0.27-1.77,0.74L8.76,3.75C8.37,3.6,7.95,3.54,7.54,3.57l2.03-2.04 c0.66-0.66,1.54-1.03,2.47-1.04c0.94,0.01,1.81,0.34,2.46,0.99C15.84,2.83,15.82,5.03,14.45,6.4z" />
    </g>
    <g>
      <path
        d="M10.07,7.48C9.84,7.63,9.53,7.56,9.38,7.33C9.29,7.19,9.19,7.07,9.07,6.95c-0.46-0.46-1.04-0.69-1.74-0.7 c-0.2,0-0.41,0.03-0.6,0.09C6.29,6.44,5.89,6.67,5.56,7l-3.3,3.3c-0.1,0.1-0.2,0.22-0.28,0.34c-0.3,0.42-0.46,0.91-0.47,1.43 c-0.01,0.66,0.24,1.28,0.7,1.74c0.96,0.96,2.53,0.94,3.51-0.04l1.52-1.52c0.39,0.15,0.81,0.21,1.22,0.18l-2.03,2.04 c-0.7,0.7-1.6,1.04-2.51,1.04c-0.88,0-1.75-0.33-2.42-0.99c-0.65-0.65-1-1.52-0.99-2.46c0.01-0.93,0.38-1.8,1.04-2.46l0.21-0.21 l3.1-3.1c0.66-0.66,1.53-1.03,2.46-1.04c0.16,0,0.31,0.01,0.46,0.03c0.76,0.09,1.46,0.42,2,0.96c0.16,0.17,0.31,0.35,0.44,0.55 C10.37,7.02,10.3,7.33,10.07,7.48z" />
    </g>
  </svg>
  
</button>
`;
